### PR TITLE
Add eus methods to command hand and gripper

### PR DESCRIPTION
**Pull request by pazeshun**
* With this PR, we can command `*baxter*`'s hand like following:
```
5.irteusgl$ send *baxter* :hand :rarm :angle-vector #f(90 90)
(90.0 90.0)
```
screenshot from 2016-11-24 13 27 37
```
6.irteusgl$ send *baxter* :hand :rarm :angle-vector #f(0 0)
(0.0 0.0)
```
screenshot from 2016-11-24 13 28 19
* Also, we can get current angle
```
7.irteusgl$ send *baxter* :hand :rarm :angle-vector #f(10 20)
(10.0 20.0)
8.irteusgl$ send *baxter* :hand :rarm :angle-vector
(10.0 20.0)
```

* We can command real hand like following:
```
4.irteusgl$ send *ri* :move-hand :rarm (send *baxter* :hand :rarm :angle-vector) 1000 :wait t
#<control_msgs::followjointtrajectoryresult #X6f65410>
```

### CODE REVIEW

**Comment by pazeshun** [comment]
```
6.irteusgl$ send *baxter* :slide-gripper :rarm 100 :relative nil
#f(0.0 20.0 -25.0 -40.0 60.0 -20.0 80.0 0.0 0.0 -20.0 -25.0 40.0 60.0 20.0 80.0 0.0 100.0 0.0)
```
screenshot from 2016-11-24 16 14 59

**Comment by pazeshun** [comment]
```
4.irteusgl$ send *baxter* :hand-grasp-pre-pose :rarm :spherical
(30.0 0.0)
```
screenshot from 2016-11-24 18 19 44
```
10.irteusgl$ send *baxter* :hand-grasp-pose :rarm :spherical  :angle 90
(30.0 90.0)
```
screenshot from 2016-11-24 18 22 23
These methods have three ways of grasping: `:pinch`, `:spherical`, `:cylindrical`

**Comment by pazeshun** [comment]
Tested in real robot

**Comment by pazeshun** [comment]
@knorth55 Please review this

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l [hunk 1]
```diff
@@ -62,6 +70,39 @@
...
+      (:pinch (send self :hand arm :angle-vector #f(0 0)))
+      (:spherical (send self :hand arm :angle-vector #f(30 0)))
+      (:cylindrical (send self :hand arm :angle-vector #f(90 0)))
+    ))
+  (:hand-grasp-pose
+    (arm style &key (angle 180))
+    (case style
+      (:pinch (send self :hand arm :angle-vector (float-vector 0 angle)))
+      (:spherical (send self :hand arm :angle-vector (float-vector 30 angle)))
+      (:cylindrical (send self :hand arm :angle-vector (float-vector 90 angle)))
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l]:
```
(t (warning-message 3 "no such style ~A~%" style))
```

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l [hunk 2]
```diff
@@ -62,6 +70,39 @@
...
+           #'(lambda (j) (send self j :joint-angle))
+           (subseq joints 0 (/ (length joints) 2)))
+         )
+       )))
+  (:hand-grasp-pre-pose
+    (arm style)
+    (case style
+      (:pinch (send self :hand arm :angle-vector #f(0 0)))
+      (:spherical (send self :hand arm :angle-vector #f(30 0)))
+      (:cylindrical (send self :hand arm :angle-vector #f(90 0)))
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l]: (t (warning-message 3 "no such style ~A~%" style))

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l [hunk 3]
```diff
@@ -62,6 +70,39 @@
     (send self arm :gripper-p :joint-angle angle :relative relative)
     (send self :angle-vector)
     )
+  (:slide-gripper
+    (arm pos &key (relative t))
+    (send self arm :gripper-x :joint-angle pos :relative relative)
+    (send self :angle-vector))
+  (:hand
+    (arm &rest args)
+    (unless args (return-from :hand nil))
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l]: add "warning-message"

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l [hunk 4]
```diff
@@ -62,6 +70,39 @@
...
+      ((:angle-vector)
+       (let ((av (cadr args)) (joints (gethash arm _hand-joints)))
+         (when av
+           (dotimes (i (length av))
+             (send self (elt joints i) :joint-angle (elt av i))
+             (send self (elt joints (+ i (length av))) :joint-angle (elt av i))))
+         (mapcar
+           #'(lambda (j) (send self j :joint-angle))
+           (subseq joints 0 (/ (length joints) 2)))
+         )
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter.l]:
```
(t (warning-message 3 "no such style ~A~%" style))
```

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter-interface.l [hunk 5]
```diff
@@ -132,6 +144,33 @@
...
+          (send traj :points
+                (list (instance trajectory_msgs::JointTrajectoryPoint
+                                :init
+                                :positions (mapcar #'deg2rad av-list)
+                                :time_from_start (ros::time (/ tm 1000)))))
+          (send goal :goal :trajectory traj)
+          (send _right-hand-action :send-goal goal)
+          (when wait
+            (send _right-hand-action :wait-for-result))
+          (send _right-hand-action :get-result)
```
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter-interface.l]: how about adding return result evaluation?
`(unless (send _right-hand-action :get-result) (warning-message 1 "failed to connect to actionlib")`

Comment (pazeshun) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter-interface.l]: `:get-result` never returns nil.
Even if connection to actionlib was failed, it returns initial `control_msgs::followjointtrajectoryresult`

**Comment by pazeshun** [comment]
I added error message outputs as they are critical errors.
Thanks @knorth55

**Comment by pazeshun** [comment]
Ping @knorth55

File: jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter-interface.l (hunk 5 above)
Comment (knorth55) [inline jsk_2016_01_baxter_apc/euslisp/jsk_2016_01_baxter_apc/baxter-interface.l]:
```
(unless (eq (send res :error_string) 0) (warining-message 3 "~~~~")
```

**Comment by pazeshun** [comment]
I added warning message to `:move-hand`

**Comment by pazeshun** [comment]
Ping @knorth55

**Pull request merged**